### ISSUE TITLE
[Dashboard] Components

### ISSUE THREAD

**Issue by ivanmalagon** [CartoDB/cartodb#14346]
### Pagination

![image](https://user-images.githubusercontent.com/1078228/47205564-909bad00-d386-11e8-849e-b932bb4644a5.png)

- Total pages
- Active page
- Show a maximum of 7 items. Always the first, ...., active - 1, active, active + 1, ..., last
     If you have 7 pages or less, show them all.
- Events: page clicked

----

### Header

There are several headers

#### Header with action buttons

![image](https://user-images.githubusercontent.com/1078228/47205654-d8223900-d386-11e8-8f77-e84f5a518cf6.png)

#### Header only with a title

![image](https://user-images.githubusercontent.com/1078228/47207096-a612d600-d38a-11e8-8f54-706600dd003f.png)

#### Header with actions and link

![image](https://user-images.githubusercontent.com/1078228/47207113-b32fc500-d38a-11e8-9464-896b762da084.png)

- Icon
- Title
- Slots (see https://vuejs.org/v2/guide/components-slots.html)
    - Link
    - Filter button
    - Action button

## Search

![image](https://user-images.githubusercontent.com/1078228/47206673-829b5b80-d389-11e8-8aaa-a3946ef45204.png)

- Search results: array of keyed links (map, dataset) all results
- Max 4 results. Try to show a candidate of each type is they exist.

### Search results header

![image](https://user-images.githubusercontent.com/1078228/47206699-9646c200-d389-11e8-9348-62598adc6d4b.png)

- Number of results and search term to compose text

----

### Map card

![image](https://user-images.githubusercontent.com/1078228/47205591-aad58b00-d386-11e8-9ea8-25a13d126d5a.png)

XL size

![image](https://user-images.githubusercontent.com/1078228/47205631-c345a580-d386-11e8-838d-9d61945408d3.png)

- Size
- Thumbnail URL (perhaps it can be formed from other properties)
- Name
- Description (or placeholder)
- Privacy
- Updated at (in plain english)
- Favorite
- Tag lists
- Selected (checked)
- Number of views
- Hovered

### Filter button

![image](https://user-images.githubusercontent.com/1078228/47205679-eb350900-d386-11e8-9155-9e9df9cbcfed.png)

- Slots
    - Show
    - Order

----

### New map button

![image](https://user-images.githubusercontent.com/1078228/47205692-f425da80-d386-11e8-969d-0f19384b9ca6.png)

- Text (it'll be shared between maps and datasets)
- Click event

----

### Quick actions

![image](https://user-images.githubusercontent.com/1078228/47205782-32bb9500-d387-11e8-8ad3-6a0ce00df590.png)

Reusable component
- Array of actions (key or event, label)

----

### Bulk actions (map... in the text it says datasets)

![image](https://user-images.githubusercontent.com/1078228/47205831-5d0d5280-d387-11e8-8617-513775435a26.png)

Reusable component
- Text (3 datasets selected, 1 map selected, 2 wadus selected)
- Array of actions

----

### Row

![image](https://user-images.githubusercontent.com/1078228/47206475-030d8c80-d389-11e8-97ad-80cc51eeff91.png)

![image](https://user-images.githubusercontent.com/1078228/47206487-0c96f480-d389-11e8-8ecf-b67bc66343aa.png)

- Title
- Type of geom
- tags
- Updated at (human readable)
- number of rows (human readable)
- size (human readable)
- privacy
- premium or not

### Dataset table

![image](https://user-images.githubusercontent.com/1078228/47210824-48838700-d394-11e8-8113-a818a153e096.png)

**Comment by jesusbotella** [comment]
We've developed almost every component in here, except the ones belonging to Search. I'm gonna close this issue, and will revisit if needed when doing Search components.
